Error while building on windows - BITNESS incorrectly detected

Hi all. First of all, sorry by advance if this post is irrelevant. I've read several related posts about building issues but still couldn't fixe my problem. Thanks also by advance for your time.

So I'm struggling to get htm.building. I followed the section of the installation tutorial 

> Simple Build On Windows (MS Visual Studio 2017 or 2019)
with this configuration:

- Windows x64
- MSVC 2017 (V 15.9)
- Cmake 3.8.2
- Python 3.7.2

The first step which happens just after clicking on startupMSVC.bat works correctly and here are the displayed logs : 
`HTM_BASE=C:\Users\Leonard\Documents\htm.core
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.17134.
-- The CXX compiler identification is MSVC 19.16.27032.1
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- REPOSITORY_DIR       = C:/Users/Leonard/Documents/htm.core
-- CMAKE_BINARY_DIR     = C:/Users/Leonard/Documents/htm.core/build/scripts
-- CMAKE_BUILD_TYPE     = Release
-- CMAKE_INSTALL_PREFIX = C:/Users/Leonard/Documents/htm.core/build/Release
-- FORCE_CPP11          = OFF
-- FORCE_BOOST          = OFF
-- BINDING_BUILD        = none
-- Environment var CXX  =

-- CMAKE_CXX_COMPILER_ID= MSVC
-- CMAKE_CXX_COMPILER_VERSION=19.16.27032.1
-- CMAKE_CXX_COMPILER   = C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe
-- CMAKE_CXX_STANDARD   = 17
-- PLATFORM             = windows
-- BITNESS              = 64
-- NEEDS_BOOST          = OFF
-- INTERNAL_CXX_FLAGS   = /permissive-;/W3;/Gm-;/EHsc;/FC;/nologo;/Zc:__cplusplus;$<$<CONFIG:Release>:/O2;/Oi;/Gy;/GL;/MD>;$<$<CONFIG:Debug>:/Ob0;/Od;/Zi;/sdl;/RTC1;/MDd>
-- INTERNAL_LINKER_FLAGS= -MACHINE:X64;-NOLOGO;-NODEFAULTLIB:LIBCMT;-ignore:4099;$<$<CONFIG:Release>:-LTCG>
-- INTERNAL_LINKER_FLAGS_STR= -MACHINE:X64 -NOLOGO -NODEFAULTLIB:LIBCMT -ignore:4099 $<$<CONFIG:Release>:-LTCG>
-- COMMON_COMPILER_DEFINITIONS_STR=_CONSOLE _MBCS NTA_OS_WINDOWS NTA_COMPILER_MSVC NTA_ARCH_64 _CRT_SECURE_NO_WARNINGS _SCL_SECURE_NO_WARNINGS _CRT_NONSTDC_NO_DEPRECATE _SCL_SECURE_NO_DEPRECATE BOOST_CONFIG_SUPPRESS_OUTDATED_MESSAGE BOOST_ALL_NO_LIB BOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS BOOST_NO_WREGEX _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING VC_EXTRALEAN WIN32_LEAN_AND_MEAN NOMINMAX NOGDI $<$<CONFIG:Debug>:NTA_ASSERTIONS_ON>
-- COMMON_OS_LIBS       = oldnames.lib;psapi.lib;ws2_32.lib
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.17134.
-- The CXX compiler identification is MSVC 19.16.27032.1
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- external import
--    REPOSITORY_DIR        = C:/Users/Leonard/Documents/htm.core
--    CMAKE_BINARY_DIR      = C:/Users/Leonard/Documents/htm.core/build/ThirdParty
--    CMAKE_INSTALL_PREFIX  = C:/Users/Leonard/Documents/htm.core/build/ThirdParty
--    CMAKE_BUILD_TYPE      = Release
--    EXTERNAL_CXX_FLAGS    = /EHsc;$<$<CONFIG:Release>:/MD>;$<$<CONFIG:Debug>:/MDd>
--    NEEDS_BOOST           = OFF
--    BINDING_BUILD         = none
--   common_INCLUDE_DIR= C:/Users/Leonard/Documents/htm.core/external/common
--   common_LIBRARIES= C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$<$<CONFIG:Release>:Release/common.lib>$<$<CONFIG:Debug>:Debug/common.lib>
-- C:/Users/Leonard/Documents/htm.core/build/ThirdParty/share/yaml-cpp.zip
-- Obtaining yaml-cpp from https://github.com/jbeder/yaml-cpp/archive/master.zip
-- The C compiler identification is MSVC 19.16.27032.1
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- obtaining Eigen
-- Obtaining gtest
-- Found PythonInterp: C:/Users/Leonard/AppData/Local/Programs/Python/Python37/python.exe (found version "3.7.2")
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE
-- obtaining MNIST data
-- Downloading/updating mnist
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.17134.
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 pour .NET Framework
Copyright (C) Microsoft Corporation. Tous droits r�serv�s.

  Checking Build System
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeFiles/generate.stamp is up-to-date.
  Creating directories for 'mnist-download'
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeFiles/generate.stamp is up-to-date.
  Performing download step (download, verify and extract) for 'mnist-download'
  -- Downloading...
     dst='C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/mnist-download-prefix/src/master.zip'
     timeout='none'
  -- Using src='https://github.com/wichtounet/mnist/archive/master.zip'
  -- Downloading... done
  -- extracting...
       src='C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/mnist-download-prefix/src/master.zip'
       dst='C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src'
  -- extracting... [tar xfz]
  -- extracting... [analysis]
  -- extracting... [rename]
  -- extracting... [clean up]
  -- extracting... done
  No skip-update step for 'mnist-download'
  No patch step for 'mnist-download'
  No configure step for 'mnist-download'
  No build step for 'mnist-download'
  No install step for 'mnist-download'
  No test step for 'mnist-download'
  Completed 'mnist-download'
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeFiles/generate.stamp is up-to-date.
-- obtaining Cereal
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/Leonard/Documents/htm.core/build/ThirdParty
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 pour .NET Framework
Copyright (C) Microsoft Corporation. Tous droits r�serv�s.

  Checking Build System
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-build/CMakeFiles/generate.stamp is up-to-date.
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/external/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
  MurmurHash3.cpp
  common.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\Release\common.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  gtest-all.cc
  gtest.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\gtest\googletest-build\googletest\Release\gtest.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  gtest_main.cc
  gtest_main.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\gtest\googletest-build\googletest\Release\gtest_main.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src/example/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-build/CMakeFiles/generate.stamp is up-to-date.
  main.cpp
c:\users\leonard\documents\htm.core\build\thirdparty\mnist_data\mnist-src\include\mnist\mnist_reader_common.hpp(46): warning C4244: 'initialisation'�: conversion de 'std::streamoff' en 'unsigned int', perte possible de donn�es [C:\Users\Leonard\Documents\htm.core\build\ThirdParty\mnist_data\mnist-build\mnist_example.vcxproj]
  mnist_example.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\mnist_data\mnist-build\Release\mnist_example.exe
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-src/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build/CMakeFiles/generate.stamp is up-to-date.
  binary.cpp
  convert.cpp
  directives.cpp
  emit.cpp
  emitfromevents.cpp
  emitter.cpp
  emitterstate.cpp
  emitterutils.cpp
  exceptions.cpp
  exp.cpp
  memory.cpp
  node.cpp
  node_data.cpp
  nodebuilder.cpp
  nodeevents.cpp
  null.cpp
  ostream_wrapper.cpp
  parse.cpp
  parser.cpp
  regex_yaml.cpp
  G�n�ration de code en cours...
  Compilation en cours...
  scanner.cpp
  scanscalar.cpp
  scantag.cpp
  scantoken.cpp
  simplekey.cpp
  singledocparser.cpp
  stream.cpp
  tag.cpp
  G�n�ration de code en cours...
  yaml-cpp.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\yaml-cpp\yaml-cpp-build\Release\libyaml-cppmd.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/external/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 pour .NET Framework
Copyright (C) Microsoft Corporation. Tous droits r�serv�s.

  Checking Build System
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-build/CMakeFiles/generate.stamp is up-to-date.
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/external/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
  MurmurHash3.cpp
  common.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\Debug\common.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  gtest-all.cc
  gtest.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\gtest\googletest-build\googletest\Debug\gtestd.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  gtest_main.cc
  gtest_main.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\gtest\googletest-build\googletest\Debug\gtest_maind.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src/example/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-build/CMakeFiles/generate.stamp is up-to-date.
  main.cpp
c:\users\leonard\documents\htm.core\build\thirdparty\mnist_data\mnist-src\include\mnist\mnist_reader_common.hpp(46): warning C4244: 'initialisation'�: conversion de 'std::streamoff' en 'unsigned int', perte possible de donn�es [C:\Users\Leonard\Documents\htm.core\build\ThirdParty\mnist_data\mnist-build\mnist_example.vcxproj]
  mnist_example.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\mnist_data\mnist-build\Debug\mnist_example.exe
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-src/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build/CMakeFiles/generate.stamp is up-to-date.
  binary.cpp
  convert.cpp
  directives.cpp
  emit.cpp
  emitfromevents.cpp
  emitter.cpp
  emitterstate.cpp
  emitterutils.cpp
  exceptions.cpp
  exp.cpp
  memory.cpp
  node.cpp
  node_data.cpp
  nodebuilder.cpp
  nodeevents.cpp
  null.cpp
  ostream_wrapper.cpp
  parse.cpp
  parser.cpp
  regex_yaml.cpp
  G�n�ration de code en cours...
  Compilation en cours...
  scanner.cpp
  scanscalar.cpp
  scantag.cpp
  scantoken.cpp
  simplekey.cpp
  singledocparser.cpp
  stream.cpp
  tag.cpp
  G�n�ration de code en cours...
  yaml-cpp.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\yaml-cpp\yaml-cpp-build\Debug\libyaml-cppmdd.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/external/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
-- Results from external build:
--   common_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/external/common
--   common_LIBRARIES = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$<$<CONFIG:Release>:Release/common.lib>$<$<CONFIG:Debug>:Debug/common.lib>
--   yaml-cpp_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-src/include
--   yaml-cpp_LIBRARIES = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build$<$<CONFIG:Release>:/Release/libyaml-cppmd.lib>$<$<CONFIG:Debug>:/Debug/libyaml-cppmdd.lib>
--   eigen_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/eigen/eigen-src
--   gtest_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/include
--   gtest_LIBRARIES = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/$<$<CONFIG:Release>:Release>$<$<CONFIG:Debug>:Debug>/gtest$<$<CONFIG:Debug>:d>.lib
--   mnist_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src/include
--   mnist_SOURCE_DIR = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src
--   cereal_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/cereal/cereal-src/include
-- Configuring htm_core src
-- MERGE_STATIC_LIBRARIES LIB_TARGET=htm_core, STATIC_LIBS = htm_core_solo;C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build$<$<CONFIG:Release>:/Release/libyaml-cppmd.lib>$<$<CONFIG:Debug>:/Debug/libyaml-cppmdd.lib>;C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$<$<CONFIG:Release>:Release/common.lib>$<$<CONFIG:Debug>:Debug/common.lib>
-- MERGE_STATIC_LIBRARIES:     htm_core_solo
-- MERGE_STATIC_LIBRARIES:     C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build$<$<CONFIG:Release>:/Release/libyaml-cppmd.lib>$<$<CONFIG:Debug>:/Debug/libyaml-cppmdd.lib> - non-target lib.
-- MERGE_STATIC_LIBRARIES:     C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$<$<CONFIG:Release>:Release/common.lib>$<$<CONFIG:Debug>:Debug/common.lib> - non-target lib.
-- Configuring htm_core tests
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/Leonard/Documents/htm.core/build/scripts
" "
You can now start Visual Studio using solution file C:\Users\Leonard\Documents\htm.core\build\scripts\htm_core.sln
Dont forget to set your default Startup Project to unit_tests.
Press any key to start Visual Studio`

So this seems to have worked fine. However, after chossing to build "Release", I get the following logs : 
'1>------ Build started: Project: ZERO_CHECK, Configuration: Release x64 ------
1>Checking Build System
1>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/CMakeFiles/generate.stamp is up-to-date.
1>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/src/CMakeFiles/generate.stamp is up-to-date.
1>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/src/test/CMakeFiles/generate.stamp is up-to-date.
2>------ Build started: Project: LibrarySource, Configuration: Release x64 ------
2>Building Custom Rule C:/Users/Leonard/Documents/htm.core/src/CMakeLists.txt
2>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/src/CMakeFiles/generate.stamp is up-to-date.
2>Anomaly.cpp
2>AnomalyLikelihood.cpp
2>Connections.cpp
2>SDRClassifier.cpp
2>SpatialPooler.cpp
2>TemporalMemory.cpp
2>c:\users\leonard\documents\htm.core\src\htm\algorithms\temporalmemory.cpp(223): warning C4267: 'argument': conversion from 'size_t' to 'htm::UInt', possible loss of data
2>ScalarEncoder.cpp
2>RandomDistributedScalarEncoder.cpp
2>Input.cpp
2>Link.cpp
2>Network.cpp
2>NuPIC.cpp
2>Output.cpp
2>Region.cpp
2>RegionImpl.cpp
2>RegionImplFactory.cpp
2>Spec.cpp
2>YAMLUtils.cpp
2>Watcher.cpp
2>ArrayBase.cpp
2>Compiling...
2>BasicType.cpp
2>Scalar.cpp
2>Value.cpp
2>Directory.cpp
2>Env.cpp
2>OS.cpp
2>OSUnix.cpp
2>OSWin.cpp
2>Path.cpp
2>Timer.cpp
2>ScalarSensor.cpp
2>SPRegion.cpp
2>TestNode.cpp
2>TMRegion.cpp
2>VectorFile.cpp
2>VectorFileEffector.cpp
2>VectorFileSensor.cpp
2>Sdr.cpp
2>LoggingException.cpp
2>LogItem.cpp
2>Compiling...
2>MovingAverage.cpp
2>Random.cpp
2>SdrMetrics.cpp
2>StlIo.cpp
2>Topology.cpp
2>LINK : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
2>LibrarySource.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\scripts\src\LibrarySource.dir\Release\LibrarySource.lib
2>Done building project "LibrarySource.vcxproj".
3>------ Build started: Project: htm_core_solo, Configuration: Release x64 ------
3>Building Custom Rule C:/Users/Leonard/Documents/htm.core/src/CMakeLists.txt
3>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/src/CMakeFiles/generate.stamp is up-to-date.
3>LINK : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
3>htm_core_solo.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\scripts\src\Release\htm_core_solo.lib
4>------ Build started: Project: htm_core, Configuration: Release x64 ------
4>Generating htm_core_dummy.c++
4>Building Custom Rule C:/Users/Leonard/Documents/htm.core/src/CMakeLists.txt
4>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/src/CMakeFiles/generate.stamp is up-to-date.
4>htm_core_dummy.c++
4>LINK : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
4>htm_core.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\scripts\src\Release\htm_core.lib
4>Combining $^<TARGET_FILE:htm_core^> for target htm_core from $^<TARGET_FILE:htm_core_solo^>;C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build$^<$^<CONFIG:Release^>:/Release/libyaml-cppmd.lib^>$^<$^<CONFIG:Debug^>:/Debug/libyaml-cppmdd.lib^>;C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$^<$^<CONFIG:Release^>:Release/common.lib^>$^<$^<CONFIG:Debug^>:Debug/common.lib^>.
4>Microsoft (R) Library Manager Version 14.16.27032.1
4>Copyright (C) Microsoft Corporation.  All rights reserved.
4>
4>htm_core_solo.lib(Anomaly.obj) : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
4>Microsoft (R) Library Manager Version 14.16.27032.1
4>Copyright (C) Microsoft Corporation.  All rights reserved.
4>
5>------ Build started: Project: benchmark_hotgym, Configuration: Release x64 ------
6>------ Build started: Project: mnist_sp, Configuration: Release x64 ------
7>------ Build started: Project: unit_tests, Configuration: Release x64 ------
5>Building Custom Rule C:/Users/Leonard/Documents/htm.core/src/CMakeLists.txt
5>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/src/CMakeFiles/generate.stamp is up-to-date.
5>Hotgym.cpp
6>Building Custom Rule C:/Users/Leonard/Documents/htm.core/src/CMakeLists.txt
6>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/src/CMakeFiles/generate.stamp is up-to-date.
7>Building Custom Rule C:/Users/Leonard/Documents/htm.core/src/test/CMakeLists.txt
7>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/src/test/CMakeFiles/generate.stamp is up-to-date.
6>MNIST_SP.cpp
7>UnitTestMain.cpp
5>HelloSPTP.cpp
7>AnomalyTest.cpp
6>c:\users\leonard\documents\htm.core\build\thirdparty\mnist_data\mnist-src\include\mnist\mnist_utils.hpp(29): warning C4244: '=': conversion from 'double' to 'unsigned char', possible loss of data
6>c:\users\leonard\documents\htm.core\build\thirdparty\mnist_data\mnist-src\include\mnist\mnist_utils.hpp(89): note: see reference to function template instantiation 'void mnist::binarize_each<std::vector<Image,std::allocator<_Ty>>>(Container &,double)' being compiled
6>        with
6>        [
6>            Image=std::vector<uint8_t,std::allocator<uint8_t>>,
6>            _Ty=std::vector<uint8_t,std::allocator<uint8_t>>,
6>            Container=std::vector<std::vector<uint8_t,std::allocator<uint8_t>>,std::allocator<std::vector<uint8_t,std::allocator<uint8_t>>>>
6>        ]
6>c:\users\leonard\documents\htm.core\src\examples\mnist\mnist_sp.cpp(106): note: see reference to function template instantiation 'void mnist::binarize_dataset<mnist::MNIST_dataset<std::vector,std::vector<uint8_t,std::allocator<_Ty>>,uint8_t>>(Dataset &)' being compiled
6>        with
6>        [
6>            _Ty=uint8_t,
6>            Dataset=mnist::MNIST_dataset<std::vector,std::vector<uint8_t,std::allocator<uint8_t>>,uint8_t>
6>        ]
6>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.16.27023\include\filesystem(2392): note: see reference to class template instantiation 'std::chrono::time_point<std::filesystem::_File_time_clock,std::filesystem::_File_time_clock::duration>' being compiled
6>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.16.27023\include\type_traits(616): note: see reference to class template instantiation 'std::basic_string_view<wchar_t,std::char_traits<wchar_t>>' being compiled
6>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.16.27023\include\xstring(2122): note: see reference to class template instantiation 'std::is_convertible<const _StringViewIsh &,std::basic_string_view<wchar_t,std::char_traits<wchar_t>>>' being compiled
6>        with
6>        [
6>            _StringViewIsh=const wchar_t *
6>        ]
6>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.16.27023\include\xstring(2122): note: see reference to variable template 'const bool conjunction_v<std::is_convertible<wchar_t const * const &,std::basic_string_view<wchar_t,std::char_traits<wchar_t> > >,std::negation<std::is_convertible<wchar_t const * const &,wchar_t const *> > >' being compiled
6>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.16.27023\include\xstring(2281): note: see reference to alias template instantiation 'std::basic_string<wchar_t,std::char_traits<wchar_t>,std::allocator<wchar_t>>::_Is_string_view_ish<_StringViewIsh>' being compiled
6>        with
6>        [
6>            _StringViewIsh=const wchar_t *
6>        ]
7>AnomalyLikelihoodTest.cpp
7>ConnectionsPerformanceTest.cpp
6>   Creating library C:/Users/Leonard/Documents/htm.core/build/scripts/src/Release/mnist_sp.lib and object C:/Users/Leonard/Documents/htm.core/build/scripts/src/Release/mnist_sp.exp
6>Generating code
5>   Creating library C:/Users/Leonard/Documents/htm.core/build/scripts/src/Release/benchmark_hotgym.lib and object C:/Users/Leonard/Documents/htm.core/build/scripts/src/Release/benchmark_hotgym.exp
5>htm_core.lib(RandomDistributedScalarEncoder.obj) : error LNK2001: unresolved external symbol "unsigned int __cdecl htm::MurmurHash3_x86_32(void const *,int,unsigned int)" (?MurmurHash3_x86_32@htm@@YAIPEBXHI@Z)
5>C:\Users\Leonard\Documents\htm.core\build\scripts\src\Release\benchmark_hotgym.exe : fatal error LNK1120: 1 unresolved externals
5>Done building project "benchmark_hotgym.vcxproj" -- FAILED.
8>------ Skipped Build: Project: hotgym, Configuration: Release x64 ------
8>Project not selected to build for this solution configuration 
6>Finished generating code
7>ConnectionsTest.cpp
6>mnist_sp.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\scripts\src\Release\mnist_sp.exe
6>Done building project "mnist_sp.vcxproj".
9>------ Skipped Build: Project: mnist, Configuration: Release x64 ------
9>Project not selected to build for this solution configuration 
7>HelloSPTPTest.cpp
7>SDRClassifierTest.cpp
7>SpatialPoolerTest.cpp
7>TemporalMemoryTest.cpp
7>ScalarEncoderTest.cpp
7>RandomDistributedScalarEncoderTest.cpp
7>CppRegionTest.cpp
7>HelloRegionTest.cpp
7>InputTest.cpp
7>LinkTest.cpp
7>NetworkTest.cpp
7>YAMLUtilsTest.cpp
7>WatcherTest.cpp
7>TopologyTest.cpp
7>ArrayTest.cpp
7>Compiling...
7>BasicTypeTest.cpp
7>CollectionTest.cpp
7>DimensionsTest.cpp
7>ScalarTest.cpp
7>ValueTest.cpp
7>DirectoryTest.cpp
7>EnvTest.cpp
7>OSTest.cpp
7>PathTest.cpp
7>TimerTest.cpp
7>RegionTestUtilities.cpp
7>SPRegionTest.cpp
7>c:\users\leonard\documents\htm.core\src\test\unit\regions\spregiontest.cpp(356): warning C4305: 'argument': truncation from 'double' to 'htm::Real32'
7>TMRegionTest.cpp
7>VectorFileTest.cpp
7>ExceptionTest.cpp
7>SdrTest.cpp
7>GroupByTest.cpp
7>MovingAverageTest.cpp
7>RandomTest.cpp
7>VectorHelpersTest.cpp
7>Compiling...
7>SdrMetricsTest.cpp
7>HelloSPTP.cpp
7>gtest.lib(gtest-all.obj) : fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'x64'
7>Done building project "unit_tests.vcxproj" -- FAILED.
10>------ Build started: Project: ALL_BUILD, Configuration: Release x64 ------
11>------ Skipped Build: Project: tests_all, Configuration: Release x64 ------
11>Project not selected to build for this solution configuration 
10>Building Custom Rule C:/Users/Leonard/Documents/htm.core/CMakeLists.txt
10>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/CMakeFiles/generate.stamp is up-to-date.
12>------ Skipped Build: Project: INSTALL, Configuration: Release x64 ------
12>Project not selected to build for this solution configuration 
13>------ Skipped Build: Project: PACKAGE, Configuration: Release x64 ------
13>Project not selected to build for this solution configuration 
========== Build: 6 succeeded, 2 failed, 0 up-to-date, 5 skipped ==========
'

As I don't know C++ and can't use MSVC thus I am unable to produce an explanation on this problem. Could someone help me fix or avoid this problem? Thank is advance!

Hi, thanks for the report! 
I'm CCing @dkeeney who is running a MS box, he'll know better with MSVC. 

> 7>gtest.lib(gtest-all.obj) : fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'x64'
7>Done building project "unit_tests.vcxproj" -- FAILED.

and 

> 7>gtest.lib(gtest-all.obj) : fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'x64'
7>Done building project "unit_tests.vcxproj" -- FAILED.

First of all, can you try a clean build? (delete everything and try again?) At a first glance it seems your system is doing a "x86" (32bit) build, while it's on x64 (64b)

> @Leonardbcm can you try the "manual" installation?
> `python setup.py install` ? OR the `cmake ../.. && make -j && make install` //you'd find more details in the Readme.

Hi and thanks for your help!
I deleted the entiere htm.core folder, then pulled again. I get this log : 
`Python version: 3.7.2 (tags/v3.7.2:9a3ffc0492, Dec 23 2018, 23:09:28) [MSC v.1916 64 bit (AMD64)]

-- Building for: Visual Studio 15 2017
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.17134.
-- The CXX compiler identification is MSVC 19.16.27032.1
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- REPOSITORY_DIR       = C:/Users/Leonard/Documents/htm.core
-- CMAKE_BINARY_DIR     = C:/Users/Leonard/Documents/htm.core/build/scripts
-- CMAKE_BUILD_TYPE     = Release
-- CMAKE_INSTALL_PREFIX = C:/Users/Leonard/Documents/htm.core/build/Release
-- FORCE_CPP11          = OFF
-- FORCE_BOOST          = OFF
-- BINDING_BUILD        = Python3
-- Environment var CXX  =

-- CMAKE_CXX_COMPILER_ID= MSVC
-- CMAKE_CXX_COMPILER_VERSION=19.16.27032.1
-- CMAKE_CXX_COMPILER   = C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- CMAKE_CXX_STANDARD   = 17
-- PLATFORM             = windows
-- BITNESS              = 32
-- NEEDS_BOOST          = OFF
-- INTERNAL_CXX_FLAGS   = /permissive-;/W3;/Gm-;/EHsc;/FC;/nologo;/Zc:__cplusplus;$<$<CONFIG:Release>:/O2;/Oi;/Gy;/GL;/MD>;$<$<CONFIG:Debug>:/Ob0;/Od;/Zi;/sdl;/RTC1;/MDd>
-- INTERNAL_LINKER_FLAGS= -MACHINE:X86;-NOLOGO;-NODEFAULTLIB:LIBCMT;-ignore:4099;$<$<CONFIG:Release>:-LTCG>
-- INTERNAL_LINKER_FLAGS_STR= -MACHINE:X86 -NOLOGO -NODEFAULTLIB:LIBCMT -ignore:4099 $<$<CONFIG:Release>:-LTCG>
-- COMMON_COMPILER_DEFINITIONS_STR=_CONSOLE _MBCS NTA_OS_WINDOWS NTA_COMPILER_MSVC NTA_ARCH_32 _CRT_SECURE_NO_WARNINGS _SCL_SECURE_NO_WARNINGS _CRT_NONSTDC_NO_DEPRECATE _SCL_SECURE_NO_DEPRECATE BOOST_CONFIG_SUPPRESS_OUTDATED_MESSAGE BOOST_ALL_NO_LIB BOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS BOOST_NO_WREGEX _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING VC_EXTRALEAN WIN32_LEAN_AND_MEAN NOMINMAX NOGDI $<$<CONFIG:Debug>:NTA_ASSERTIONS_ON>
-- COMMON_OS_LIBS       = oldnames.lib;psapi.lib;ws2_32.lib
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.17134.
-- The CXX compiler identification is MSVC 19.16.27032.1
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- external import
--    REPOSITORY_DIR        = C:/Users/Leonard/Documents/htm.core
--    CMAKE_BINARY_DIR      = C:/Users/Leonard/Documents/htm.core/build/ThirdParty
--    CMAKE_INSTALL_PREFIX  = C:/Users/Leonard/Documents/htm.core/build/ThirdParty
--    CMAKE_BUILD_TYPE      = Release
--    EXTERNAL_CXX_FLAGS    = /EHsc;$<$<CONFIG:Release>:/MD>;$<$<CONFIG:Debug>:/MDd>
--    NEEDS_BOOST           = OFF
--    BINDING_BUILD         = Python3
--   common_INCLUDE_DIR= C:/Users/Leonard/Documents/htm.core/external/common
--   common_LIBRARIES= C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$<$<CONFIG:Release>:Release/common.lib>$<$<CONFIG:Debug>:Debug/common.lib>
-- C:/Users/Leonard/Documents/htm.core/build/ThirdParty/share/yaml-cpp.zip
-- Obtaining yaml-cpp from https://github.com/jbeder/yaml-cpp/archive/master.zip
-- The C compiler identification is MSVC 19.16.27032.1
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- obtaining Eigen
-- Obtaining gtest
-- Found PythonInterp: C:/Users/Leonard/AppData/Local/Programs/Python/Python37/python.exe (found version "3.7.2")
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE
-- obtaining PyBind11
-- obtaining MNIST data
-- Downloading/updating mnist
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.17134.
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 pour .NET Framework
Copyright (C) Microsoft Corporation. Tous droits r�serv�s.

  Checking Build System
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeFiles/generate.stamp is up-to-date.
  Creating directories for 'mnist-download'
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeFiles/generate.stamp is up-to-date.
  Performing download step (download, verify and extract) for 'mnist-download'
  -- Downloading...
     dst='C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/mnist-download-prefix/src/master.zip'
     timeout='none'
  -- Using src='https://github.com/wichtounet/mnist/archive/master.zip'
  -- Downloading... done
  -- extracting...
       src='C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/mnist-download-prefix/src/master.zip'
       dst='C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src'
  -- extracting... [tar xfz]
  -- extracting... [analysis]
  -- extracting... [rename]
  -- extracting... [clean up]
  -- extracting... done
  No skip-update step for 'mnist-download'
  No patch step for 'mnist-download'
  No configure step for 'mnist-download'
  No build step for 'mnist-download'
  No install step for 'mnist-download'
  No test step for 'mnist-download'
  Completed 'mnist-download'
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeFiles/generate.stamp is up-to-date.
-- obtaining Cereal
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/Leonard/Documents/htm.core/build/ThirdParty
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 pour .NET Framework
Copyright (C) Microsoft Corporation. Tous droits r�serv�s.

  Checking Build System
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-build/CMakeFiles/generate.stamp is up-to-date.
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/external/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
  MurmurHash3.cpp
  common.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\Release\common.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  gtest-all.cc
  gtest.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\gtest\googletest-build\googletest\Release\gtest.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  gtest_main.cc
  gtest_main.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\gtest\googletest-build\googletest\Release\gtest_main.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src/example/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-build/CMakeFiles/generate.stamp is up-to-date.
  main.cpp
c:\users\leonard\documents\htm.core\build\thirdparty\mnist_data\mnist-src\include\mnist\mnist_reader_common.hpp(46): warning C4244: 'initializing': conversion from 'std::streamoff' to 'unsigned int', possible loss of data [C:\Users\Leonard\Documents\htm.core\build\ThirdParty\mnist_data\mnist-build\mnist_example.vcxproj]
  mnist_example.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\mnist_data\mnist-build\Release\mnist_example.exe
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-src/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build/CMakeFiles/generate.stamp is up-to-date.
  binary.cpp
  convert.cpp
  directives.cpp
  emit.cpp
  emitfromevents.cpp
  emitter.cpp
  emitterstate.cpp
  emitterutils.cpp
  exceptions.cpp
  exp.cpp
  memory.cpp
  node.cpp
  node_data.cpp
  nodebuilder.cpp
  nodeevents.cpp
  null.cpp
  ostream_wrapper.cpp
  parse.cpp
  parser.cpp
  regex_yaml.cpp
  Generating Code...
  Compiling...
  scanner.cpp
  scanscalar.cpp
  scantag.cpp
  scantoken.cpp
  simplekey.cpp
  singledocparser.cpp
  stream.cpp
  tag.cpp
  Generating Code...
  yaml-cpp.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\yaml-cpp\yaml-cpp-build\Release\libyaml-cppmd.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/external/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 pour .NET Framework
Copyright (C) Microsoft Corporation. Tous droits r�serv�s.

  Checking Build System
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-build/CMakeFiles/generate.stamp is up-to-date.
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/external/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
  MurmurHash3.cpp
  common.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\Debug\common.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  gtest-all.cc
  gtest.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\gtest\googletest-build\googletest\Debug\gtestd.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  gtest_main.cc
  gtest_main.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\gtest\googletest-build\googletest\Debug\gtest_maind.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src/example/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-build/CMakeFiles/generate.stamp is up-to-date.
  main.cpp
c:\users\leonard\documents\htm.core\build\thirdparty\mnist_data\mnist-src\include\mnist\mnist_reader_common.hpp(46): warning C4244: 'initializing': conversion from 'std::streamoff' to 'unsigned int', possible loss of data [C:\Users\Leonard\Documents\htm.core\build\ThirdParty\mnist_data\mnist-build\mnist_example.vcxproj]
  mnist_example.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\mnist_data\mnist-build\Debug\mnist_example.exe
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-src/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build/CMakeFiles/generate.stamp is up-to-date.
  binary.cpp
  convert.cpp
  directives.cpp
  emit.cpp
  emitfromevents.cpp
  emitter.cpp
  emitterstate.cpp
  emitterutils.cpp
  exceptions.cpp
  exp.cpp
  memory.cpp
  node.cpp
  node_data.cpp
  nodebuilder.cpp
  nodeevents.cpp
  null.cpp
  ostream_wrapper.cpp
  parse.cpp
  parser.cpp
  regex_yaml.cpp
  Generating Code...
  Compiling...
  scanner.cpp
  scanscalar.cpp
  scantag.cpp
  scantoken.cpp
  simplekey.cpp
  singledocparser.cpp
  stream.cpp
  tag.cpp
  Generating Code...
  yaml-cpp.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\yaml-cpp\yaml-cpp-build\Debug\libyaml-cppmdd.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/external/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
-- Results from external build:
--   common_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/external/common
--   common_LIBRARIES = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$<$<CONFIG:Release>:Release/common.lib>$<$<CONFIG:Debug>:Debug/common.lib>
--   yaml-cpp_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-src/include
--   yaml-cpp_LIBRARIES = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build$<$<CONFIG:Release>:/Release/libyaml-cppmd.lib>$<$<CONFIG:Debug>:/Debug/libyaml-cppmdd.lib>
--   eigen_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/eigen/eigen-src
--   gtest_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/include
--   gtest_LIBRARIES = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/$<$<CONFIG:Release>:Release>$<$<CONFIG:Debug>:Debug>/gtest$<$<CONFIG:Debug>:d>.lib
--   pybind11_SOURCE_DIR = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/pybind11/pybind11-src
--   pybind11_BINARY_DIR = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/pybind11/pybind11-build
--   mnist_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src/include
--   mnist_SOURCE_DIR = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src
--   cereal_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/cereal/cereal-src/include
-- Configuring htm_core src
-- MERGE_STATIC_LIBRARIES LIB_TARGET=htm_core, STATIC_LIBS = htm_core_solo;C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build$<$<CONFIG:Release>:/Release/libyaml-cppmd.lib>$<$<CONFIG:Debug>:/Debug/libyaml-cppmdd.lib>;C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$<$<CONFIG:Release>:Release/common.lib>$<$<CONFIG:Debug>:Debug/common.lib>
-- MERGE_STATIC_LIBRARIES:     htm_core_solo
-- MERGE_STATIC_LIBRARIES:     C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build$<$<CONFIG:Release>:/Release/libyaml-cppmd.lib>$<$<CONFIG:Debug>:/Debug/libyaml-cppmdd.lib> - non-target lib.
-- MERGE_STATIC_LIBRARIES:     C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$<$<CONFIG:Release>:Release/common.lib>$<$<CONFIG:Debug>:Debug/common.lib> - non-target lib.
-- Configuring htm_core tests
-- Configuring Python interface
-- Found PythonInterp: C:/Users/Leonard/AppData/Local/Programs/Python/Python37/python.exe (found version "3.7.2")
CMake Error at build/ThirdParty/pybind11/pybind11-src/tools/FindPythonLibsNew.cmake:122 (message):
  Python config failure: Python is 64-bit, chosen compiler is 32-bit
Call Stack (most recent call first):
  build/ThirdParty/pybind11/pybind11-src/tools/pybind11Tools.cmake:16 (find_package)
  build/ThirdParty/pybind11/pybind11-src/CMakeLists.txt:33 (include)


-- Configuring incomplete, errors occurred!
See also "C:/Users/Leonard/Documents/htm.core/build/scripts/CMakeFiles/CMakeOutput.log".
Traceback (most recent call last):
  File "setup.py", line 53, in <module>

  File "C:\Users\Leonard\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "C:\Users\Leonard\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "C:\Users\Leonard\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\Leonard\Documents\htm.core\bindings\py\packaging\setup.py", line 224, in <module>
    getExtensionFiles(platform)
  File "C:\Users\Leonard\Documents\htm.core\bindings\py\packaging\setup.py", line 178, in getExtensionFiles
    generateExtensions()
  File "C:\Users\Leonard\Documents\htm.core\bindings\py\packaging\setup.py", line 209, in generateExtensions
    subprocess.check_call(["cmake", PY_VER, REPO_DIR])
  File "C:\Users\Leonard\AppData\Local\Programs\Python\Python37\lib\subprocess.py", line 347, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '-DBINDING_BUILD=Python3', 'C:\\Users\\Leonard\\Documents\\htm.core']' returned non-zero exit status 1.`


It seems from the first lines : 
`-- Building for: Visual Studio 15 2017
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.17134.
-- The CXX compiler identification is MSVC 19.16.27032.1
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works`

that MSVC chose a x86 compiler as a default compiler. This strangely does not happend when I use startupMSVC.bat which produce (after re-deleting and cloning the whole code): 
`C:\Users\Leonard\Documents\htm.core>startupMSVC.bat
HTM_BASE=C:\Users\Leonard\Documents\htm.core
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.17134.
-- The CXX compiler identification is MSVC 19.16.27032.1
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- REPOSITORY_DIR       = C:/Users/Leonard/Documents/htm.core
-- CMAKE_BINARY_DIR     = C:/Users/Leonard/Documents/htm.core/build/scripts
-- CMAKE_BUILD_TYPE     = Release
-- CMAKE_INSTALL_PREFIX = C:/Users/Leonard/Documents/htm.core/build/Release
-- FORCE_CPP11          = OFF
-- FORCE_BOOST          = OFF
-- BINDING_BUILD        = none
-- Environment var CXX  =

-- CMAKE_CXX_COMPILER_ID= MSVC
-- CMAKE_CXX_COMPILER_VERSION=19.16.27032.1
-- CMAKE_CXX_COMPILER   = C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe
-- CMAKE_CXX_STANDARD   = 17
-- PLATFORM             = windows
-- BITNESS              = 64
-- NEEDS_BOOST          = OFF
-- INTERNAL_CXX_FLAGS   = /permissive-;/W3;/Gm-;/EHsc;/FC;/nologo;/Zc:__cplusplus;$<$<CONFIG:Release>:/O2;/Oi;/Gy;/GL;/MD>;$<$<CONFIG:Debug>:/Ob0;/Od;/Zi;/sdl;/RTC1;/MDd>
-- INTERNAL_LINKER_FLAGS= -MACHINE:X64;-NOLOGO;-NODEFAULTLIB:LIBCMT;-ignore:4099;$<$<CONFIG:Release>:-LTCG>
-- INTERNAL_LINKER_FLAGS_STR= -MACHINE:X64 -NOLOGO -NODEFAULTLIB:LIBCMT -ignore:4099 $<$<CONFIG:Release>:-LTCG>
-- COMMON_COMPILER_DEFINITIONS_STR=_CONSOLE _MBCS NTA_OS_WINDOWS NTA_COMPILER_MSVC NTA_ARCH_64 _CRT_SECURE_NO_WARNINGS _SCL_SECURE_NO_WARNINGS _CRT_NONSTDC_NO_DEPRECATE _SCL_SECURE_NO_DEPRECATE BOOST_CONFIG_SUPPRESS_OUTDATED_MESSAGE BOOST_ALL_NO_LIB BOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS BOOST_NO_WREGEX _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING VC_EXTRALEAN WIN32_LEAN_AND_MEAN NOMINMAX NOGDI $<$<CONFIG:Debug>:NTA_ASSERTIONS_ON>
-- COMMON_OS_LIBS       = oldnames.lib;psapi.lib;ws2_32.lib
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.17134.
-- The CXX compiler identification is MSVC 19.16.27032.1
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- external import
--    REPOSITORY_DIR        = C:/Users/Leonard/Documents/htm.core
--    CMAKE_BINARY_DIR      = C:/Users/Leonard/Documents/htm.core/build/ThirdParty
--    CMAKE_INSTALL_PREFIX  = C:/Users/Leonard/Documents/htm.core/build/ThirdParty
--    CMAKE_BUILD_TYPE      = Release
--    EXTERNAL_CXX_FLAGS    = /EHsc;$<$<CONFIG:Release>:/MD>;$<$<CONFIG:Debug>:/MDd>
--    NEEDS_BOOST           = OFF
--    BINDING_BUILD         = none
--   common_INCLUDE_DIR= C:/Users/Leonard/Documents/htm.core/external/common
--   common_LIBRARIES= C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$<$<CONFIG:Release>:Release/common.lib>$<$<CONFIG:Debug>:Debug/common.lib>
-- C:/Users/Leonard/Documents/htm.core/build/ThirdParty/share/yaml-cpp.zip
-- Obtaining yaml-cpp from https://github.com/jbeder/yaml-cpp/archive/master.zip
-- The C compiler identification is MSVC 19.16.27032.1
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- obtaining Eigen
-- Obtaining gtest
-- Found PythonInterp: C:/Users/Leonard/AppData/Local/Programs/Python/Python37/python.exe (found version "3.7.2")
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE
-- obtaining MNIST data
-- Downloading/updating mnist
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.17134.
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 pour .NET Framework
Copyright (C) Microsoft Corporation. Tous droits r�serv�s.

  Checking Build System
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeFiles/generate.stamp is up-to-date.
  Creating directories for 'mnist-download'
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeFiles/generate.stamp is up-to-date.
  Performing download step (download, verify and extract) for 'mnist-download'
  -- Downloading...
     dst='C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/mnist-download-prefix/src/master.zip'
     timeout='none'
  -- Using src='https://github.com/wichtounet/mnist/archive/master.zip'
  -- Downloading... done
  -- extracting...
       src='C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/mnist-download-prefix/src/master.zip'
       dst='C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src'
  -- extracting... [tar xfz]
  -- extracting... [analysis]
  -- extracting... [rename]
  -- extracting... [clean up]
  -- extracting... done
  No skip-update step for 'mnist-download'
  No patch step for 'mnist-download'
  No configure step for 'mnist-download'
  No build step for 'mnist-download'
  No install step for 'mnist-download'
  No test step for 'mnist-download'
  Completed 'mnist-download'
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeFiles/generate.stamp is up-to-date.
-- obtaining Cereal
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/Leonard/Documents/htm.core/build/ThirdParty
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 pour .NET Framework
Copyright (C) Microsoft Corporation. Tous droits r�serv�s.

  Checking Build System
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-build/CMakeFiles/generate.stamp is up-to-date.
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/external/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
  MurmurHash3.cpp
  common.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\Release\common.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  gtest-all.cc
  gtest.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\gtest\googletest-build\googletest\Release\gtest.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  gtest_main.cc
  gtest_main.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\gtest\googletest-build\googletest\Release\gtest_main.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src/example/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-build/CMakeFiles/generate.stamp is up-to-date.
  main.cpp
c:\users\leonard\documents\htm.core\build\thirdparty\mnist_data\mnist-src\include\mnist\mnist_reader_common.hpp(46): warning C4244: 'initializing': conversion from 'std::streamoff' to 'unsigned int', possible loss of data [C:\Users\Leonard\Documents\htm.core\build\ThirdParty\mnist_data\mnist-build\mnist_example.vcxproj]
  mnist_example.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\mnist_data\mnist-build\Release\mnist_example.exe
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-src/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build/CMakeFiles/generate.stamp is up-to-date.
  binary.cpp
  convert.cpp
  directives.cpp
  emit.cpp
  emitfromevents.cpp
  emitter.cpp
  emitterstate.cpp
  emitterutils.cpp
  exceptions.cpp
  exp.cpp
  memory.cpp
  node.cpp
  node_data.cpp
  nodebuilder.cpp
  nodeevents.cpp
  null.cpp
  ostream_wrapper.cpp
  parse.cpp
  parser.cpp
  regex_yaml.cpp
  Generating Code...
  Compiling...
  scanner.cpp
  scanscalar.cpp
  scantag.cpp
  scantoken.cpp
  simplekey.cpp
  singledocparser.cpp
  stream.cpp
  tag.cpp
  Generating Code...
  yaml-cpp.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\yaml-cpp\yaml-cpp-build\Release\libyaml-cppmd.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/external/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 pour .NET Framework
Copyright (C) Microsoft Corporation. Tous droits r�serv�s.

  Checking Build System
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-build/CMakeFiles/generate.stamp is up-to-date.
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/external/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
  MurmurHash3.cpp
  common.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\Debug\common.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  gtest-all.cc
  gtest.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\gtest\googletest-build\googletest\Debug\gtestd.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  gtest_main.cc
  gtest_main.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\gtest\googletest-build\googletest\Debug\gtest_maind.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src/example/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-build/CMakeFiles/generate.stamp is up-to-date.
  main.cpp
c:\users\leonard\documents\htm.core\build\thirdparty\mnist_data\mnist-src\include\mnist\mnist_reader_common.hpp(46): warning C4244: 'initializing': conversion from 'std::streamoff' to 'unsigned int', possible loss of data [C:\Users\Leonard\Documents\htm.core\build\ThirdParty\mnist_data\mnist-build\mnist_example.vcxproj]
  mnist_example.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\mnist_data\mnist-build\Debug\mnist_example.exe
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-src/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build/CMakeFiles/generate.stamp is up-to-date.
  binary.cpp
  convert.cpp
  directives.cpp
  emit.cpp
  emitfromevents.cpp
  emitter.cpp
  emitterstate.cpp
  emitterutils.cpp
  exceptions.cpp
  exp.cpp
  memory.cpp
  node.cpp
  node_data.cpp
  nodebuilder.cpp
  nodeevents.cpp
  null.cpp
  ostream_wrapper.cpp
  parse.cpp
  parser.cpp
  regex_yaml.cpp
  Generating Code...
  Compiling...
  scanner.cpp
  scanscalar.cpp
  scantag.cpp
  scantoken.cpp
  simplekey.cpp
  singledocparser.cpp
  stream.cpp
  tag.cpp
  Generating Code...
  yaml-cpp.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\yaml-cpp\yaml-cpp-build\Debug\libyaml-cppmdd.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/external/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
-- Results from external build:
--   common_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/external/common
--   common_LIBRARIES = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$<$<CONFIG:Release>:Release/common.lib>$<$<CONFIG:Debug>:Debug/common.lib>
--   yaml-cpp_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-src/include
--   yaml-cpp_LIBRARIES = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build$<$<CONFIG:Release>:/Release/libyaml-cppmd.lib>$<$<CONFIG:Debug>:/Debug/libyaml-cppmdd.lib>
--   eigen_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/eigen/eigen-src
--   gtest_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/include
--   gtest_LIBRARIES = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/$<$<CONFIG:Release>:Release>$<$<CONFIG:Debug>:Debug>/gtest$<$<CONFIG:Debug>:d>.lib
--   mnist_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src/include
--   mnist_SOURCE_DIR = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src
--   cereal_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/cereal/cereal-src/include
-- Configuring htm_core src
-- MERGE_STATIC_LIBRARIES LIB_TARGET=htm_core, STATIC_LIBS = htm_core_solo;C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build$<$<CONFIG:Release>:/Release/libyaml-cppmd.lib>$<$<CONFIG:Debug>:/Debug/libyaml-cppmdd.lib>;C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$<$<CONFIG:Release>:Release/common.lib>$<$<CONFIG:Debug>:Debug/common.lib>
-- MERGE_STATIC_LIBRARIES:     htm_core_solo
-- MERGE_STATIC_LIBRARIES:     C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build$<$<CONFIG:Release>:/Release/libyaml-cppmd.lib>$<$<CONFIG:Debug>:/Debug/libyaml-cppmdd.lib> - non-target lib.
-- MERGE_STATIC_LIBRARIES:     C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$<$<CONFIG:Release>:Release/common.lib>$<$<CONFIG:Debug>:Debug/common.lib> - non-target lib.
-- Configuring htm_core tests
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/Leonard/Documents/htm.core/build/scripts
" "
You can now start Visual Studio using solution file C:\Users\Leonard\Documents\htm.core\build\scripts\htm_core.sln
Dont forget to set your default Startup Project to unit_tests.
Press any key to start Visual Studio`

and the first lines : 

`C:\Users\Leonard\Documents\htm.core>startupMSVC.bat
HTM_BASE=C:\Users\Leonard\Documents\htm.core
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.17134.
-- The CXX compiler identification is MSVC 19.16.27032.1
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe -- works`

are different than the "manual" build using setup.py. Here it looks like the good compiler has been chosen.

However, this does not fix the problem since I still get this log while building the entiere project in MSVC:

`1>------ Build started: Project: ZERO_CHECK, Configuration: Release x64 ------
1>Checking Build System
1>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/CMakeFiles/generate.stamp is up-to-date.
1>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/src/CMakeFiles/generate.stamp is up-to-date.
1>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/src/test/CMakeFiles/generate.stamp is up-to-date.
2>------ Build started: Project: LibrarySource, Configuration: Release x64 ------
2>Building Custom Rule C:/Users/Leonard/Documents/htm.core/src/CMakeLists.txt
2>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/src/CMakeFiles/generate.stamp is up-to-date.
2>Anomaly.cpp
2>AnomalyLikelihood.cpp
2>Connections.cpp
2>SDRClassifier.cpp
2>SpatialPooler.cpp
2>TemporalMemory.cpp
2>c:\users\leonard\documents\htm.core\src\htm\algorithms\temporalmemory.cpp(223): warning C4267: 'argument': conversion from 'size_t' to 'htm::UInt', possible loss of data
2>ScalarEncoder.cpp
2>RandomDistributedScalarEncoder.cpp
2>Input.cpp
2>Link.cpp
2>Network.cpp
2>NuPIC.cpp
2>Output.cpp
2>Region.cpp
2>RegionImpl.cpp
2>RegionImplFactory.cpp
2>Spec.cpp
2>YAMLUtils.cpp
2>Watcher.cpp
2>ArrayBase.cpp
2>Compiling...
2>BasicType.cpp
2>Scalar.cpp
2>Value.cpp
2>Directory.cpp
2>Env.cpp
2>OS.cpp
2>OSUnix.cpp
2>OSWin.cpp
2>Path.cpp
2>Timer.cpp
2>ScalarSensor.cpp
2>SPRegion.cpp
2>TestNode.cpp
2>TMRegion.cpp
2>VectorFile.cpp
2>VectorFileEffector.cpp
2>VectorFileSensor.cpp
2>Sdr.cpp
2>LoggingException.cpp
2>LogItem.cpp
2>Compiling...
2>MovingAverage.cpp
2>Random.cpp
2>SdrMetrics.cpp
2>StlIo.cpp
2>Topology.cpp
2>LINK : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
2>LibrarySource.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\scripts\src\LibrarySource.dir\Release\LibrarySource.lib
2>Done building project "LibrarySource.vcxproj".
3>------ Build started: Project: htm_core_solo, Configuration: Release x64 ------
3>Building Custom Rule C:/Users/Leonard/Documents/htm.core/src/CMakeLists.txt
3>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/src/CMakeFiles/generate.stamp is up-to-date.
3>LINK : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
3>htm_core_solo.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\scripts\src\Release\htm_core_solo.lib
4>------ Build started: Project: htm_core, Configuration: Release x64 ------
4>Generating htm_core_dummy.c++
4>Building Custom Rule C:/Users/Leonard/Documents/htm.core/src/CMakeLists.txt
4>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/src/CMakeFiles/generate.stamp is up-to-date.
4>htm_core_dummy.c++
4>LINK : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
4>htm_core.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\scripts\src\Release\htm_core.lib
4>Combining $^<TARGET_FILE:htm_core^> for target htm_core from $^<TARGET_FILE:htm_core_solo^>;C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build$^<$^<CONFIG:Release^>:/Release/libyaml-cppmd.lib^>$^<$^<CONFIG:Debug^>:/Debug/libyaml-cppmdd.lib^>;C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$^<$^<CONFIG:Release^>:Release/common.lib^>$^<$^<CONFIG:Debug^>:Debug/common.lib^>.
4>Microsoft (R) Library Manager Version 14.16.27032.1
4>Copyright (C) Microsoft Corporation.  All rights reserved.
4>
4>htm_core_solo.lib(Anomaly.obj) : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
4>Microsoft (R) Library Manager Version 14.16.27032.1
4>Copyright (C) Microsoft Corporation.  All rights reserved.
4>
5>------ Build started: Project: benchmark_hotgym, Configuration: Release x64 ------
6>------ Build started: Project: mnist_sp, Configuration: Release x64 ------
7>------ Build started: Project: unit_tests, Configuration: Release x64 ------
5>Building Custom Rule C:/Users/Leonard/Documents/htm.core/src/CMakeLists.txt
5>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/src/CMakeFiles/generate.stamp is up-to-date.
5>Hotgym.cpp
6>Building Custom Rule C:/Users/Leonard/Documents/htm.core/src/CMakeLists.txt
6>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/src/CMakeFiles/generate.stamp is up-to-date.
7>Building Custom Rule C:/Users/Leonard/Documents/htm.core/src/test/CMakeLists.txt
7>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/src/test/CMakeFiles/generate.stamp is up-to-date.
6>MNIST_SP.cpp
7>UnitTestMain.cpp
5>HelloSPTP.cpp
7>AnomalyTest.cpp
6>c:\users\leonard\documents\htm.core\build\thirdparty\mnist_data\mnist-src\include\mnist\mnist_utils.hpp(29): warning C4244: '=': conversion from 'double' to 'unsigned char', possible loss of data
6>c:\users\leonard\documents\htm.core\build\thirdparty\mnist_data\mnist-src\include\mnist\mnist_utils.hpp(89): note: see reference to function template instantiation 'void mnist::binarize_each<std::vector<Image,std::allocator<_Ty>>>(Container &,double)' being compiled
6>        with
6>        [
6>            Image=std::vector<uint8_t,std::allocator<uint8_t>>,
6>            _Ty=std::vector<uint8_t,std::allocator<uint8_t>>,
6>            Container=std::vector<std::vector<uint8_t,std::allocator<uint8_t>>,std::allocator<std::vector<uint8_t,std::allocator<uint8_t>>>>
6>        ]
6>c:\users\leonard\documents\htm.core\src\examples\mnist\mnist_sp.cpp(106): note: see reference to function template instantiation 'void mnist::binarize_dataset<mnist::MNIST_dataset<std::vector,std::vector<uint8_t,std::allocator<_Ty>>,uint8_t>>(Dataset &)' being compiled
6>        with
6>        [
6>            _Ty=uint8_t,
6>            Dataset=mnist::MNIST_dataset<std::vector,std::vector<uint8_t,std::allocator<uint8_t>>,uint8_t>
6>        ]
6>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.16.27023\include\filesystem(2392): note: see reference to class template instantiation 'std::chrono::time_point<std::filesystem::_File_time_clock,std::filesystem::_File_time_clock::duration>' being compiled
6>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.16.27023\include\type_traits(616): note: see reference to class template instantiation 'std::basic_string_view<wchar_t,std::char_traits<wchar_t>>' being compiled
6>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.16.27023\include\xstring(2122): note: see reference to class template instantiation 'std::is_convertible<const _StringViewIsh &,std::basic_string_view<wchar_t,std::char_traits<wchar_t>>>' being compiled
6>        with
6>        [
6>            _StringViewIsh=const wchar_t *
6>        ]
6>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.16.27023\include\xstring(2122): note: see reference to variable template 'const bool conjunction_v<std::is_convertible<wchar_t const * const &,std::basic_string_view<wchar_t,std::char_traits<wchar_t> > >,std::negation<std::is_convertible<wchar_t const * const &,wchar_t const *> > >' being compiled
6>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.16.27023\include\xstring(2281): note: see reference to alias template instantiation 'std::basic_string<wchar_t,std::char_traits<wchar_t>,std::allocator<wchar_t>>::_Is_string_view_ish<_StringViewIsh>' being compiled
6>        with
6>        [
6>            _StringViewIsh=const wchar_t *
6>        ]
7>AnomalyLikelihoodTest.cpp
7>ConnectionsPerformanceTest.cpp
6>   Creating library C:/Users/Leonard/Documents/htm.core/build/scripts/src/Release/mnist_sp.lib and object C:/Users/Leonard/Documents/htm.core/build/scripts/src/Release/mnist_sp.exp
6>Generating code
5>   Creating library C:/Users/Leonard/Documents/htm.core/build/scripts/src/Release/benchmark_hotgym.lib and object C:/Users/Leonard/Documents/htm.core/build/scripts/src/Release/benchmark_hotgym.exp
5>htm_core.lib(RandomDistributedScalarEncoder.obj) : error LNK2001: unresolved external symbol "unsigned int __cdecl htm::MurmurHash3_x86_32(void const *,int,unsigned int)" (?MurmurHash3_x86_32@htm@@YAIPEBXHI@Z)
5>C:\Users\Leonard\Documents\htm.core\build\scripts\src\Release\benchmark_hotgym.exe : fatal error LNK1120: 1 unresolved externals
5>Done building project "benchmark_hotgym.vcxproj" -- FAILED.
8>------ Skipped Build: Project: hotgym, Configuration: Release x64 ------
8>Project not selected to build for this solution configuration 
7>ConnectionsTest.cpp
6>Finished generating code
6>mnist_sp.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\scripts\src\Release\mnist_sp.exe
6>Done building project "mnist_sp.vcxproj".
9>------ Skipped Build: Project: mnist, Configuration: Release x64 ------
9>Project not selected to build for this solution configuration 
7>HelloSPTPTest.cpp
7>SDRClassifierTest.cpp
7>SpatialPoolerTest.cpp
7>TemporalMemoryTest.cpp
7>ScalarEncoderTest.cpp
7>RandomDistributedScalarEncoderTest.cpp
7>CppRegionTest.cpp
7>HelloRegionTest.cpp
7>InputTest.cpp
7>LinkTest.cpp
7>NetworkTest.cpp
7>YAMLUtilsTest.cpp
7>WatcherTest.cpp
7>TopologyTest.cpp
7>ArrayTest.cpp
7>Compiling...
7>BasicTypeTest.cpp
7>CollectionTest.cpp
7>DimensionsTest.cpp
7>ScalarTest.cpp
7>ValueTest.cpp
7>DirectoryTest.cpp
7>EnvTest.cpp
7>OSTest.cpp
7>PathTest.cpp
7>TimerTest.cpp
7>RegionTestUtilities.cpp
7>SPRegionTest.cpp
7>c:\users\leonard\documents\htm.core\src\test\unit\regions\spregiontest.cpp(356): warning C4305: 'argument': truncation from 'double' to 'htm::Real32'
7>TMRegionTest.cpp
7>VectorFileTest.cpp
7>ExceptionTest.cpp
7>SdrTest.cpp
7>GroupByTest.cpp
7>MovingAverageTest.cpp
7>RandomTest.cpp
7>VectorHelpersTest.cpp
7>Compiling...
7>SdrMetricsTest.cpp
7>HelloSPTP.cpp
7>gtest.lib(gtest-all.obj) : fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'x64'
7>Done building project "unit_tests.vcxproj" -- FAILED.
10>------ Build started: Project: ALL_BUILD, Configuration: Release x64 ------
11>------ Skipped Build: Project: tests_all, Configuration: Release x64 ------
11>Project not selected to build for this solution configuration 
10>Building Custom Rule C:/Users/Leonard/Documents/htm.core/CMakeLists.txt
10>CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/scripts/CMakeFiles/generate.stamp is up-to-date.
12>------ Skipped Build: Project: INSTALL, Configuration: Release x64 ------
12>Project not selected to build for this solution configuration 
13>------ Skipped Build: Project: PACKAGE, Configuration: Release x64 ------
13>Project not selected to build for this solution configuration 
========== Build: 6 succeeded, 2 failed, 0 up-to-date, 5 skipped ==========`

Thanks.

> > Moreover, I never used Cmake, what command line/arguments should I use after cmake -DBITNESS=64 -DCMAKE_BUILD_TYPE=Release ../..?
> 
> nothing, that's it. Assuming you're in `REPO/build/scripts/` (the 2 folders can be created manually), the `../..` should point to the repo root REPO (where CMakeLists.txt is)

It seems to have worked well. Here are the logs : 
`
C:\Users\Leonard\Documents\htm.core\build\scripts>cmake -DBITNESS=64 -DCMAKE_BUILD_TYPE=Release ../..
-- Building for: Visual Studio 15 2017
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.17134.
-- The CXX compiler identification is MSVC 19.16.27032.1
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- REPOSITORY_DIR       = C:/Users/Leonard/Documents/htm.core
-- CMAKE_BINARY_DIR     = C:/Users/Leonard/Documents/htm.core/build/scripts
-- CMAKE_BUILD_TYPE     = Release
-- CMAKE_INSTALL_PREFIX = C:/Users/Leonard/Documents/htm.core/build/Release
-- FORCE_CPP11          = OFF
-- FORCE_BOOST          = OFF
-- BINDING_BUILD        = none
-- Environment var CXX  =

-- CMAKE_CXX_COMPILER_ID= MSVC
-- CMAKE_CXX_COMPILER_VERSION=19.16.27032.1
-- CMAKE_CXX_COMPILER   = C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- CMAKE_CXX_STANDARD   = 17
-- PLATFORM             = windows
-- BITNESS              = 64
-- NEEDS_BOOST          = OFF
-- INTERNAL_CXX_FLAGS   = /permissive-;/W3;/Gm-;/EHsc;/FC;/nologo;/Zc:__cplusplus;$<$<CONFIG:Release>:/O2;/Oi;/Gy;/GL;/MD>;$<$<CONFIG:Debug>:/Ob0;/Od;/Zi;/sdl;/RTC1;/MDd>
-- INTERNAL_LINKER_FLAGS= -MACHINE:X64;-NOLOGO;-NODEFAULTLIB:LIBCMT;-ignore:4099;$<$<CONFIG:Release>:-LTCG>
-- INTERNAL_LINKER_FLAGS_STR= -MACHINE:X64 -NOLOGO -NODEFAULTLIB:LIBCMT -ignore:4099 $<$<CONFIG:Release>:-LTCG>
-- COMMON_COMPILER_DEFINITIONS_STR=_CONSOLE _MBCS NTA_OS_WINDOWS NTA_COMPILER_MSVC NTA_ARCH_64 _CRT_SECURE_NO_WARNINGS _SCL_SECURE_NO_WARNINGS _CRT_NONSTDC_NO_DEPRECATE _SCL_SECURE_NO_DEPRECATE BOOST_CONFIG_SUPPRESS_OUTDATED_MESSAGE BOOST_ALL_NO_LIB BOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS BOOST_NO_WREGEX _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING VC_EXTRALEAN WIN32_LEAN_AND_MEAN NOMINMAX NOGDI $<$<CONFIG:Debug>:NTA_ASSERTIONS_ON>
-- COMMON_OS_LIBS       = oldnames.lib;psapi.lib;ws2_32.lib
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.17134.
-- The CXX compiler identification is MSVC 19.16.27032.1
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- external import
--    REPOSITORY_DIR        = C:/Users/Leonard/Documents/htm.core
--    CMAKE_BINARY_DIR      = C:/Users/Leonard/Documents/htm.core/build/ThirdParty
--    CMAKE_INSTALL_PREFIX  = C:/Users/Leonard/Documents/htm.core/build/ThirdParty
--    CMAKE_BUILD_TYPE      = Release
--    EXTERNAL_CXX_FLAGS    = /EHsc;$<$<CONFIG:Release>:/MD>;$<$<CONFIG:Debug>:/MDd>
--    NEEDS_BOOST           = OFF
--    BINDING_BUILD         = none
--   common_INCLUDE_DIR= C:/Users/Leonard/Documents/htm.core/external/common
--   common_LIBRARIES= C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$<$<CONFIG:Release>:Release/common.lib>$<$<CONFIG:Debug>:Debug/common.lib>
-- C:/Users/Leonard/Documents/htm.core/build/ThirdParty/share/yaml-cpp.zip
-- Obtaining yaml-cpp from https://github.com/jbeder/yaml-cpp/archive/master.zip
-- The C compiler identification is MSVC 19.16.27032.1
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- obtaining Eigen
-- Obtaining gtest
-- Found PythonInterp: C:/Users/Leonard/AppData/Local/Programs/Python/Python37/python.exe (found version "3.7.2")
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE
-- obtaining MNIST data
-- Downloading/updating mnist
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.17134.
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 pour .NET Framework
Copyright (C) Microsoft Corporation. Tous droits r�serv�s.

  Checking Build System
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeFiles/generate.stamp is up-to-date.
  Creating directories for 'mnist-download'
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeFiles/generate.stamp is up-to-date.
  Performing download step (download, verify and extract) for 'mnist-download'
  -- Downloading...
     dst='C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/mnist-download-prefix/src/master.zip'
     timeout='none'
  -- Using src='https://github.com/wichtounet/mnist/archive/master.zip'
  -- Downloading... done
  -- extracting...
       src='C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/mnist-download-prefix/src/master.zip'
       dst='C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src'
  -- extracting... [tar xfz]
  -- extracting... [analysis]
  -- extracting... [rename]
  -- extracting... [clean up]
  -- extracting... done
  No skip-update step for 'mnist-download'
  No patch step for 'mnist-download'
  No configure step for 'mnist-download'
  No build step for 'mnist-download'
  No install step for 'mnist-download'
  No test step for 'mnist-download'
  Completed 'mnist-download'
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-download/CMakeFiles/generate.stamp is up-to-date.
-- obtaining Cereal
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/Leonard/Documents/htm.core/build/ThirdParty
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 pour .NET Framework
Copyright (C) Microsoft Corporation. Tous droits r�serv�s.

  Checking Build System
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-build/CMakeFiles/generate.stamp is up-to-date.
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/external/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
  MurmurHash3.cpp
  common.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\Release\common.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  gtest-all.cc
  gtest.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\gtest\googletest-build\googletest\Release\gtest.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  gtest_main.cc
  gtest_main.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\gtest\googletest-build\googletest\Release\gtest_main.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src/example/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-build/CMakeFiles/generate.stamp is up-to-date.
  main.cpp
c:\users\leonard\documents\htm.core\build\thirdparty\mnist_data\mnist-src\include\mnist\mnist_reader_common.hpp(46): warning C4244: 'initializing': conversion from 'std::streamoff' to 'unsigned int', possible loss of data [C:\Users\Leonard\Documents\htm.core\build\ThirdParty\mnist_data\mnist-build\mnist_example.vcxproj]
  mnist_example.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\mnist_data\mnist-build\Release\mnist_example.exe
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-src/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build/CMakeFiles/generate.stamp is up-to-date.
  binary.cpp
  convert.cpp
  directives.cpp
  emit.cpp
  emitfromevents.cpp
  emitter.cpp
  emitterstate.cpp
  emitterutils.cpp
  exceptions.cpp
  exp.cpp
  memory.cpp
  node.cpp
  node_data.cpp
  nodebuilder.cpp
  nodeevents.cpp
  null.cpp
  ostream_wrapper.cpp
  parse.cpp
  parser.cpp
  regex_yaml.cpp
  Generating Code...
  Compiling...
  scanner.cpp
  scanscalar.cpp
  scantag.cpp
  scantoken.cpp
  simplekey.cpp
  singledocparser.cpp
  stream.cpp
  tag.cpp
  Generating Code...
  yaml-cpp.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\yaml-cpp\yaml-cpp-build\Release\libyaml-cppmd.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/external/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 pour .NET Framework
Copyright (C) Microsoft Corporation. Tous droits r�serv�s.

  Checking Build System
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-build/CMakeFiles/generate.stamp is up-to-date.
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/external/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
  MurmurHash3.cpp
  common.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\Debug\common.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  gtest-all.cc
  gtest.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\gtest\googletest-build\googletest\Debug\gtestd.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
  gtest_main.cc
  gtest_main.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\gtest\googletest-build\googletest\Debug\gtest_maind.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src/example/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-build/CMakeFiles/generate.stamp is up-to-date.
  main.cpp
c:\users\leonard\documents\htm.core\build\thirdparty\mnist_data\mnist-src\include\mnist\mnist_reader_common.hpp(46): warning C4244: 'initializing': conversion from 'std::streamoff' to 'unsigned int', possible loss of data [C:\Users\Leonard\Documents\htm.core\build\ThirdParty\mnist_data\mnist-build\mnist_example.vcxproj]
  mnist_example.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\mnist_data\mnist-build\Debug\mnist_example.exe
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-src/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build/CMakeFiles/generate.stamp is up-to-date.
  binary.cpp
  convert.cpp
  directives.cpp
  emit.cpp
  emitfromevents.cpp
  emitter.cpp
  emitterstate.cpp
  emitterutils.cpp
  exceptions.cpp
  exp.cpp
  memory.cpp
  node.cpp
  node_data.cpp
  nodebuilder.cpp
  nodeevents.cpp
  null.cpp
  ostream_wrapper.cpp
  parse.cpp
  parser.cpp
  regex_yaml.cpp
  Generating Code...
  Compiling...
  scanner.cpp
  scanscalar.cpp
  scantag.cpp
  scantoken.cpp
  simplekey.cpp
  singledocparser.cpp
  stream.cpp
  tag.cpp
  Generating Code...
  yaml-cpp.vcxproj -> C:\Users\Leonard\Documents\htm.core\build\ThirdParty\yaml-cpp\yaml-cpp-build\Debug\libyaml-cppmdd.lib
  Building Custom Rule C:/Users/Leonard/Documents/htm.core/external/CMakeLists.txt
  CMake does not need to re-run because C:/Users/Leonard/Documents/htm.core/build/ThirdParty/CMakeFiles/generate.stamp is up-to-date.
-- Results from external build:
--   common_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/external/common
--   common_LIBRARIES = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$<$<CONFIG:Release>:Release/common.lib>$<$<CONFIG:Debug>:Debug/common.lib>
--   yaml-cpp_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-src/include
--   yaml-cpp_LIBRARIES = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build$<$<CONFIG:Release>:/Release/libyaml-cppmd.lib>$<$<CONFIG:Debug>:/Debug/libyaml-cppmdd.lib>
--   eigen_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/eigen/eigen-src
--   gtest_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-src/googletest/include
--   gtest_LIBRARIES = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/gtest/googletest-build/googletest/$<$<CONFIG:Release>:Release>$<$<CONFIG:Debug>:Debug>/gtest$<$<CONFIG:Debug>:d>.lib
--   mnist_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src/include
--   mnist_SOURCE_DIR = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/mnist_data/mnist-src
--   cereal_INCLUDE_DIRS = C:/Users/Leonard/Documents/htm.core/build/ThirdParty/cereal/cereal-src/include
-- Configuring htm_core src
-- MERGE_STATIC_LIBRARIES LIB_TARGET=htm_core, STATIC_LIBS = htm_core_solo;C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build$<$<CONFIG:Release>:/Release/libyaml-cppmd.lib>$<$<CONFIG:Debug>:/Debug/libyaml-cppmdd.lib>;C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$<$<CONFIG:Release>:Release/common.lib>$<$<CONFIG:Debug>:Debug/common.lib>
-- MERGE_STATIC_LIBRARIES:     htm_core_solo
-- MERGE_STATIC_LIBRARIES:     C:/Users/Leonard/Documents/htm.core/build/ThirdParty/yaml-cpp/yaml-cpp-build$<$<CONFIG:Release>:/Release/libyaml-cppmd.lib>$<$<CONFIG:Debug>:/Debug/libyaml-cppmdd.lib> - non-target lib.
-- MERGE_STATIC_LIBRARIES:     C:/Users/Leonard/Documents/htm.core/build/ThirdParty/$<$<CONFIG:Release>:Release/common.lib>$<$<CONFIG:Debug>:Debug/common.lib> - non-target lib.
-- Configuring htm_core tests
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/Leonard/Documents/htm.core/build/scripts`

I am realy unfamiliar with C++, what is the next step for using the python bindings now? Thanks

> @Leonardbcm now compile the c++:
> 
> > make -j8 && make install
> 
I'm very sorry but this does not work. It seems that the commande "make" isn't defined. I installed GnuWin32 which provide a make commande. 
Here what's happening:
`C:\Users\Leonard\Documents\htm.core\build\scripts>make -j8 && make install
make: *** Pas de cibles spécifiées et aucun makefile n'a été trouvé. Arrêt.`

Which is "No target specified and no makefile found. Stop" in en. Note that I'm using this command from build\scripts

> But @Leonardbcm seems to have a different problem. I assume he is running a 64bit Windows.

Yes I am using a 64bit Windows.

--- a/bindings/py/packaging/setup.py
+++ b/bindings/py/packaging/setup.py
@@ -175,14 +175,14 @@ def getExtensionFiles(platform):
   files = getExtensionFileNames(platform)
   for f in files:
     if not os.path.exists(f):
-      generateExtensions()
+      generateExtensions(platform)
       break
 
   return files
 
 
 
-def generateExtensions():
+def generateExtensions(platform):
   """
   This will perform a full Release build with default arguments.
   The CMake build will copy everything in the Repository/bindings/py/packaging 
@@ -206,7 +206,10 @@ def generateExtensions():
     if not os.path.isdir(scriptsDir):
       os.makedirs(scriptsDir)
     os.chdir(scriptsDir)
-    subprocess.check_call(["cmake", PY_VER, REPO_DIR])
+    if platform == WINDOWS_PLATFORMS:
+      subprocess.check_call(["cmake", PY_VER, REPO_DIR, "-A", "x64"])
+    else:
+      subprocess.check_call(["cmake", PY_VER, REPO_DIR])
     subprocess.check_call(["cmake", "--build", ".", "--target", "install", "--config", "Release"])
   finally:
     os.chdir(cwd)

--- a/bindings/py/packaging/setup.py
+++ b/bindings/py/packaging/setup.py
@@ -188,6 +188,7 @@ def generateExtensions(platform):
   The CMake build will copy everything in the Repository/bindings/py/packaging 
   directory to the distr directory (Repository/build/Release/distr)
   and then create the extension libraries in Repository/build/Release/distr/src/nupic/bindings.
+  Note: for Windows it will force a X64 build.
   """
   cwd = os.getcwd()
   

--- a/src/htm/algorithms/TemporalMemory.cpp
+++ b/src/htm/algorithms/TemporalMemory.cpp
@@ -220,7 +220,7 @@ static void growSynapses(Connections &connections,
   const size_t nActualWithMax = std::min(nActual, static_cast<size_t>(maxSynapsesPerSegment) - connections.numSynapses(segment));
 
   // Pick nActual cells randomly.
-  for (const auto syn : rng.sample(candidates, nActualWithMax)) {
+  for (const auto syn : rng.sample(candidates, static_cast<UInt>(nActualWithMax))) {
     connections.createSynapse(segment, syn, initialPermanence); //TODO createSynapse consider creating a vector of new synapses at once?
   }
 }

